api/jobs/handlers.py

```diff
@@ -175,14 +175,17 @@ def post(self, cid):
 
         doc = self.request.json
 
-        validate_data(doc, 'rule-add.json', 'input', 'POST', optional=True)
+        validate_data(doc, 'rule-new.json', 'input', 'POST', optional=True)
         try:
-            get_gear_by_name(doc['alg'])
+            gear = get_gear_by_name(doc['alg'])
         except APINotFoundException:
             self.abort(400, 'Cannot find gear for alg {}, alg not valid'.format(doc['alg']))
 
         doc['project_id'] = cid
 
+        if 'config' in doc:
+            validate_gear_config(gear, fill_gear_default_values(gear, doc['config']))
```

Comment:
This should be called unconditionally, setting `doc['config']` to {} and calling `fill_gear_default_values` on it.